yjit/src/codegen.rs

```diff
@@ -3924,50 +3922,62 @@ fn gen_send_iseq(jit: &JITState, ctx: &Context, ci: * const rb_callinfo, cme: *
     }
 
     // Number of locals that are not parameters
-    const int num_locals = iseq->body->local_table_size - num_params;
+    let num_locals = unsafe { get_iseq_body_local_table_size(iseq) as i32 } - num_params;
 
     // Create a side-exit to fall back to the interpreter
-    uint8_t *side_exit = get_side_exit(jit, ocb, ctx);
+    let side_exit = get_side_exit(jit, ocb, ctx);
 
     // Check for interrupts
     gen_check_ints(cb, side_exit);
 
-    const struct rb_builtin_function *leaf_builtin = rb_leaf_builtin_function(iseq);
-
-    if (leaf_builtin && !block && leaf_builtin->argc + 1 /* for self */ + 1 /* for ec */ <= NUM_C_ARG_REGS) {
-        add_comment(cb, "inlined leaf builtin");
-
-        // Call the builtin func (ec, recv, arg1, arg2, ...)
-        mov(cb, C_ARG_REGS[0], REG_EC);
-
-        // Copy self and arguments
-        for (int32_t i = 0; i < leaf_builtin->argc + 1; i++) {
-            let stack_opnd = ctx.stack_opnd(leaf_builtin->argc - i);
-            let c_arg_reg = C_ARG_REGS[i + 1];
-            mov(cb, c_arg_reg, stack_opnd);
-        }
-        ctx.stack_pop(leaf_builtin->argc + 1);
-        call_ptr(cb, REG0, (void *)leaf_builtin->func_ptr);
+    let leaf_builtin = rb_leaf_builtin_function(iseq);
+    match (block, leaf_builtin) {
+        (None, Some(builtin_info)) => {
+            let builtin_argc = unsafe { get_builtin_argc(builtin_info) };
+            if builtin_argc + 1 /* for self */ + 1 /* for ec */ <= (C_ARG_REGS.len() as i32) {
+                add_comment(cb, "inlined leaf builtin");
+
+                // Call the builtin func (ec, recv, arg1, arg2, ...)
+                mov(cb, C_ARG_REGS[0], REG_EC);
+
+                // Copy self and arguments
+                for i in 0..=builtin_argc {
```

Comment:
I think this is also missing the +1 from the above loop end condition

Reply:
The two you comment on both have a "..=" instead of a "..". That's a Rust inclusive range, so it should (I believe) also hit the top element. https://doc.rust-lang.org/nightly/std/ops/struct.RangeInclusive.html

Reply:
Did not know about the inclusive range 👍